core/store/src/db/rocksdb.rs

```diff
@@ -525,23 +524,27 @@ impl RocksDB {
         instance_tracker::block_until_all_instances_are_dropped();
     }
 
-    /// Returns version of the database state on disk.
-    pub(crate) fn get_version(path: &Path, config: &StoreConfig) -> io::Result<DbVersion> {
-        let db = RocksDB::open(path, config, Mode::ReadOnly)?;
-        let value =
-            db.get_raw_bytes(DBCol::DbVersion, crate::db::VERSION_KEY)?.ok_or_else(|| {
-                other_error(
-                    "Failed to read database version; \
-                     it’s not a neard database or database is corrupted."
-                        .into(),
-                )
-            })?;
-        serde_json::from_slice(&value).map_err(|_err| {
-            other_error(format!(
-                "Failed to parse database version: {value:?}; \
+    /// Returns version of the database state on disk.  Returns `None` if the
+    /// database does not exist.
+    pub(crate) fn get_version(path: &Path, config: &StoreConfig) -> io::Result<Option<DbVersion>> {
+        if !path.join("CURRENT").is_file() {
+            return Ok(None);
+        }
+
+        // Specify only DBCol::DbVersion because it’s ok to open database in
+        // read-only mode without specifying all column families the database
+        // contains but it’s an error to provide a descriptor for column family
+        // which does note exist.
```

Comment:
```suggestion
        // which does not exist.
```

Reply:
This comment could further justify why avoiding this error is desirable (e.g. making it possible to read out the database version regardless of which version the current database is at, thus avoiding errors before the migrations have concluded)